### PR TITLE
[SPARK-45504][SS] Lower CPU Priority Of RocksDB Background Threads

### PR DESCRIPTION
**Pull request by siying**
### What changes were proposed in this pull request?
Lower RocksDB background threads' CPU priority.

### Why are the changes needed?
RocksDB background threads execute background tasks such as compactions and flushes. It should be in general lower priority than Spark tasks excution. For the case where a task may wait for some RocksDB background task to finish, such as checkpointing, or waiting async checkpointing to finish, the task slot is waiting so we are likely to have enough CPU for low pri CPU anyway.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Run benchmarks and see the results.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala [hunk 1]
```diff
@@ -109,6 +109,8 @@ class RocksDB(
   dbOptions.setCreateIfMissing(true)
   dbOptions.setTableFormatConfig(tableFormatConfig)
   dbOptions.setMaxOpenFiles(conf.maxOpenFiles)
+  dbOptions.getEnv().lowerThreadPoolCPUPriority(Priority.HIGH)
```
Comment (HeartSaVioR) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala:112]: Just a curiosity, should we set this as up and down? Or unnecessary line?

**Review by HeartSaVioR — COMMENTED** [review]
Shall we put some performance check result here? We need some prove for rationalization of this change.

**Comment by siying** [comment]
The benchmark results not decisive. Sometimes it does show P99 improvements, but P50 tends to regress. Closing it for now. Will reopen if things change.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala (hunk 1 above)
Comment (siying) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala:112]: This cannot set up or down. Once priority for a thread is lowered, there is no way to move it up.

**Pull request closed without merging**